elements/lisk-p2p/test/functional/message_rate.ts

```diff
@@ -224,4 +225,127 @@ describe('Message rate limit', () => {
+
+			// Act
+			// Send nodeInfo message 3 times, so total of 4 messages of max allowed and not get ban score
+			for (let i = 0; i < 3; i += 1) {
+				attackerNode.applyNodeInfo(attackerNode.nodeInfo);
+			}
+
+			await wait(10);
+
+			// Assert
+			expect(targetNode['_peerBook'].triedPeers[0].internalState?.reputation).toEqual(100);
+		});
+
+		it('should add a ban score of peer when it sends postNodeInfo messages more than 4 within 10 seconds', async () => {
+			// Arrange
+			const attackerNode = p2pNodeList[0];
+			const targetNode = p2pNodeList[1];
+
+			// Act
+			// Send nodeInfo message 4 times and once its send on connection to get a ban score
+			for (let i = 0; i < 4; i += 1) {
+				attackerNode.applyNodeInfo(attackerNode.nodeInfo);
+			}
+
+			await wait(10);
+
+			// Assert
+			expect(targetNode['_peerBook'].triedPeers[0].internalState?.reputation).toEqual(90);
+		});
+
+		it('should ban a peer when it sends postNodeInfo messages more than 4 + 10 times within 10 seconds', async () => {
+			// Arrange
+			const attackerNode = p2pNodeList[0];
+			const targetNode = p2pNodeList[1];
+
+			// Act
+			// Send nodeInfo message 14 times and once its send on connection (total of 15 messages to get banned)
+			for (let i = 0; i < 14; i += 1) {
+				attackerNode.applyNodeInfo(attackerNode.nodeInfo);
+			}
+
+			await wait(10);
+
+			// Assert
+			expect(removedPeers.get(targetNode.config.port)).toEqual(
+				expect.arrayContaining([attackerNode.config.port.toString()]),
+			);
+		});
+	});
+
+	describe('Request peerList or nodeInfo more than once', () => {
+		it('should ban a peer when it requests getPeers RPC explicitly for more than 10 times', async () => {
+			// Arrange
+			const attackerNode = p2pNodeList[0];
+			const targetNode = p2pNodeList[1];
+
+			const targetPeerId = `127.0.0.1:${targetNode.config.port}`;
+
+			// Act
+			// With every getPeers RPC request apply penalty of 10, 10 * 10 times will result in banning
+			for (let i = 0; i < 10; i += 1) {
+				await wait(1);
+				// eslint-disable-next-line @typescript-eslint/no-floating-promises
+				(async () => {
+					try {
+						await attackerNode.requestFromPeer(
+							{
+								procedure: REMOTE_EVENT_RPC_GET_PEERS_LIST,
+							},
+							targetPeerId,
+						);
+						// eslint-disable-next-line no-empty
+					} catch (error) {}
+				})();
+			}
+
+			await wait(10);
+
+			// Assert
+			expect(removedPeers.get(targetNode.config.port)).toEqual(
+				expect.arrayContaining([attackerNode.config.port.toString()]),
+			);
+		});
+
+		it('should ban a peer when it requests RPC getNodeInfo explicitly for more than 10 times', async () => {
+			// Arrange
+			const attackerNode = p2pNodeList[0];
+			const targetNode = p2pNodeList[1];
+
+			const targetPeerId = `127.0.0.1:${targetNode.config.port}`;
+
+			// Act
+			// With every getNodeInfo RPC request apply penalty of 10, 10 * 10 times will result in banning
+			for (let i = 0; i < 10; i += 1) {
```

Comment:
If request is allowed one time without penalty, should this be `i < 11`?

Reply:
when a peer connects, it asks for status and peer list immediately, so, one request is already done while connecting and when we do 10 other requests it gets banned